fat/fat2/transactionio_test.go

```diff
@@ -0,0 +1,120 @@
+package fat2_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	. "github.com/Factom-Asset-Tokens/fatd/fat/fat2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Table to drive the tests. JSON strings are used for future portability of test cases
+// to languages other than Go
+var transactionIOUnmarshalJSONTests = []struct {
+	Name  string
+	Error string
+	JSON  string
+}{{
+	Name: "valid (address, type, amount)",
+	JSON: `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","type":"PEG","amount":50}`,
+}, {
+	Name: "valid (allow zero value)",
+	JSON: `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","type":"PEG","amount":0}`,
+}, {
+	Name: "valid (type, amount)",
+	JSON: `{"type":"PEG","amount":50}`,
+}, {
+	Name: "valid (address, amount)",
+	JSON: `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","amount":50}`,
+}, {
+	Name: "valid (address, type)",
+	JSON: `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","type":"PEG"}`,
+}, {
+	Name:  "invalid (address)",
+	Error: "checksum error",
+	JSON:  `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6Rbad","type":"PEG","amount":50}`,
+}, {
+	Name:  "invalid (negative amount)",
+	Error: "cannot unmarshal number",
+	JSON:  `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","type":"PEG","amount":-50}`,
+}, {
+	Name:  "invalid (non-string type)",
+	Error: "cannot unmarshal number",
+	JSON:  `{"address":"FA2jK2HcLnRdS94dEcU27rF3meoJfpUcZPSinpb7AwQvPRY6RL1Q","type":50,"amount":50}`,
+}}
+
+func TestTransactionIO_UnmarshalJSON(t *testing.T) {
```

Comment:
Thank you for pointing this out. I had originally written the test because I planned to write a custom `UnmarshalJSON()` but then when it came to implementation, I opted for the simpler route of using the default unmarshaller and pushing validation to a separate function.

Removing the tests now